contracts/WitnetRequestsBoard.sol

```diff
@@ -597,4 +602,16 @@ contract WitnetRequestsBoard is WitnetRequestsBoardInterface {
   function getBlockNumber() internal view virtual returns (uint256) {
     return block.number;
   }
+
+  /// @dev Returns the relayer address or the requester address in case the block inclusion was already paid
+  /// @param _id The unique identifier of the data request.
+  /// @param _blockHash The hash of the block which inclusion has to be rewarded.
+  function getAddressBlockReward(uint256 _id, uint256 _blockHash) internal returns (address) {
+      if (paidBlocks[_blockHash] == true) {
+          return requests[_id].requestor;
+      } else {
+          paidBlocks[_blockHash] = true;
+          return blockRelay.readRelayerAddress(_blockHash, requests[_id].epoch);
+      }
+  }
```

Comment:
We always entertained the idea of having 1 Block Relay serving multiple Witnet Request Boards. Under this assumption, it may be possible that a block relayer receives more than 1 reward from different WRB instances.